src/shims/sync.rs

```diff
@@ -377,7 +413,9 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 Ok(0)
             } else {
                 // Trying to acquire the same mutex again.
-                if kind == this.eval_libc("PTHREAD_MUTEX_NORMAL")? {
+                if is_mutex_kind_default(this, kind)? {
+                    throw_ub_format!("trying to acquire already locked PTHREAD_MUTEX_DEFAULT");
```

Comment:
Please add a test case for this case.

Reply:
Probably better two, to be sure: one where we pass a NULL attribute, and once where we initialize the attribute but do not `settype` it.

Oh, and do wehave a testcase that when we initialize with `NORMAL`, we get the deadlock error as expected?